search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt

```diff
@@ -307,6 +307,31 @@ class TemporalEntityAttributeService(
             entity[it.key] = it.value
         }
 
+        emptyTemporalAttributesArrays(entity, withTemporalValues)
         return JsonLdEntity(entity, jsonLdEntity.contexts)
     }
+
+    private fun emptyTemporalAttributesArrays(entity: MutableMap<String, Any>, withTemporalValues: Boolean) {
+        // temporal attributes must have empty arrays if their injected temporal evolutions are empty
+        entity.filter {
+            it.value is List<*> &&
+                (it.value as List<*>).size == 1
```

Comment:
I don't really get what is happening here.
Is this the check to know if a temporal evolution is empty ?

Reply:
I am checking if there is a temporal attribute left having the property NGSILD_PROPERTY_VALUE that should have been removed. Such temporal attribute should have NGSILD_PROPERTY_VALUES instead